musicplatform/users/views.py

```diff
@@ -0,0 +1,28 @@
+from django.shortcuts import render
+from rest_framework import viewsets, permissions, status, mixins
+from rest_framework.response import Response
+from .models import ExUser
+from .serializers import ExUserSerializer
+
+# Create your views here.
+class ExUserViewSet(mixins.RetrieveModelMixin,
+                    mixins.UpdateModelMixin,
+                    mixins.DestroyModelMixin,
+                    mixins.ListModelMixin,
+                    viewsets.GenericViewSet):
+    
+    queryset = ExUser.objects.all()
+    serializer_class = ExUserSerializer
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly]
+    
+    def delete(self, request, *args, **kwargs):
+        if (request.user.username!='admin') and (self.kwargs['pk'] != str(request.user.pk)):
```

Comment:
instead of checking like that you can create a 'permissions.py` file and add the following class to it:
```python
class UserPermission(permissions.BasePermission):
    def has_object_permission(self, request, view, obj):
        # Read permissions are allowed to any request,
        # so we'll always allow GET, HEAD or OPTIONS requests.
        if request.method in permissions.SAFE_METHODS:
            return True

        # Instance must have an attribute named `owner`.
        return obj == request.user
```

it checks if the user making the request is the same user or not,  then you can add it to your views permission_classes